ntbs-service/Services/NotificationChangesService.cs

```diff
@@ -86,106 +85,87 @@ private IEnumerable<AuditLog> GetCanonicalLogs(List<AuditLog> group)
             {
-                foreach (var item in group)
-                {
-                    yield return item;
-                }
-
-                yield break;
+                return group;
             }
 
             // Social risks page contains sub-models, which create their own audit entries
             var socialRisksUpdate = group.Find(log => log.EntityType == nameof(SocialRiskFactors));
             if (socialRisksUpdate != null)
             {
-                yield return socialRisksUpdate;
-                yield break;
+                return new List<AuditLog> {socialRisksUpdate};
             }
 
             // Travel and visitor details get saved together, but it actually makes sense to show them independently
             if (group.Count == 2
                 && group.Any(log => log.EntityType == nameof(TravelDetails))
                 && group.Any(log => log.EntityType == nameof(VisitorDetails)))
             {
-                foreach (var item in group)
-                {
-                    yield return item;
-                }
-
-                yield break;
+                return group;
             }
 
             // Transferring creates a flurry of related events.
             // For each of these events we just want to have a single item in the history view 
-            var transferAlertLog = group.Find(log => log.EntityType == nameof(TransferAlert)
-                                                     || log.EntityType == nameof(TransferRejectedAlert));
-            if (transferAlertLog != null)
+
+            // Rejection creates:
+            //   - TransferAlert - Update
+            //   - TransferRejectedAlert - Insert
+            if (group.Any(log => log.EntityType == nameof(TransferRejectedAlert)))
             {
-                // Rejection creates:
-                //   - TransferAlert - Update
-                //   - TransferRejectedAlert - Insert
-                if (group.Any(log => log.EntityType == nameof(TransferRejectedAlert)))
+                var rejectionAlertLog = group.Find(log => log.EntityType == nameof(TransferRejectedAlert));
+                // but we need to filter out the subsequent closure of the rejection alert!
+                if (rejectionAlertLog.EventType == "Insert")
                 {
-                    var rejectionAlertLog = group.Find(log => log.EntityType == nameof(TransferRejectedAlert));
-                    // but we need to filter out the subsequent closure of the rejection alert!
-                    if (rejectionAlertLog.EventType == "Insert")
-                    {
-                        rejectionAlertLog.ActionString = "rejected";
-                        yield return rejectionAlertLog;
-                    }
+                    rejectionAlertLog.ActionString = "rejected";
+                    return new List<AuditLog> {rejectionAlertLog};
                 }
+
+                return new List<AuditLog>();
```

Comment:
It's probably worth adding a comment here saying that when a `TransferRejectedAlert` is cancelled, we don't want to display any logs.